src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -101,13 +121,71 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
       docToUpdate.set(DocumentConstants.Fields.DOC_ID, replaceDocumentId);
     }
     docToUpdate.setAll(replaceDocument());
+    //    // restore the original $vectorize
+    //    docToUpdate.put(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD,
+    // vectorizeNode.asText());
     // return modified flag as true
     return true;
   }
 
+  /**
```

Comment:
This is the second level of update as I mentioned above.
will try to vectorize as needed and update the $vector accordingly